migration/src/main/scala/akka/persistence/r2dbc/migration/MigrationTool.scala

```diff
@@ -93,7 +93,7 @@ object MigrationTool {
  *
  * Note: tags are not migrated.
  */
-class MigrationTool(system: ActorSystem[_]) {
+class MigrationTool(system: ActorSystem[_], entityTypeExtractor: String => String = PersistenceId.extractEntityType) {
```

Comment:
I don't think this will help much, because `PersistenceId.extractEntityType` is used all over the place when persisting events and durable state. It's based on the assumption that the default `PersistenceId` from Akka Typed is used.

In general I think a migration tool like this should be seen as an example that can be adjusted to application specific needs by modifying the source code (beauty of OSS).

If we want to make the binary more flexible I think a better solution is to be able to define a mapping function for the EventEnvelope from the source (sourceEventsByPersistenceIdQuery). Then it would be possible to rewrite the persistenceId for example. That is already possible by writing your own `CurrentEventsByPersistenceIdQuery`, but that is maybe too complicated?

Reply:
entityType is used for eventsBySlices query, yes